integrations/GA/browser.js

```diff
@@ -1,122 +1,986 @@
-        : undefined;
-
-    if (path) {
-      ga("rudder_ga.set", "page", path);
-    }
+    if (this.classic) {
+      var props = rudderElement.message.properties;
+      var category = props.category;
+      var name = props.name;
+      window._gaq.push("_trackPageview", path(this.props, this.includeSearch));
 
-    if (title) {
-      ga("rudder_ga.set", "title", title);
-    }
+      if ((category && this, this.trackCategorizedPages)) {
+        this.track(rudderElement, { nonInteraction: 1 });
+      }
+
+      if (name && this.trackNamedPages) {
+        this.track(rudderElement, { nonInteraction: 1 });
+      }
+    } else {
+      var dimensionsArray = {};
+      for (let val of this.dimensions) {
+        dimensionsArray[val.from] = val.to;
+      }
+      var metricsArray = {};
+      for (let val of this.metrics) {
+        metricsArray[val.from] = val.to;
+      }
+      var contentGroupingsArray = {};
+      for (let val of this.contentGroupings) {
+        contentGroupingsArray[val.from] = val.to;
+      }
+      logger.debug("in GoogleAnalyticsManager page");
+
+      var category = rudderElement.message.properties.category;
+      var eventProperties = rudderElement.message.properties;
+      var name =
+        rudderElement.message.properties.category +
+        " " +
+        rudderElement.message.name;
+      var campaign = rudderElement.message.context.campaign | {};
+      var pageview = {};
+      var pagePath = path(eventProperties, this.includeSearch);
+      var pageReferrer = rudderElement.message.properties.referrer || "";
+      var pageTitle;
+      if (
+        !rudderElement.message.properties.category &&
+        !rudderElement.message.name
+      )
+        pageTitle = eventProperties.title;
+      else if (!rudderElement.message.properties.category)
+        pageTitle = rudderElement.message.name;
+      else if (!rudderElement.message.name)
+        pageTitle = rudderElement.message.properties.category;
+      else pageTitle = name;
+
+      pageview.page = pagePath;
+      pageview.title = pageTitle;
+      pageview.location = eventProperties.url;
+
+      if (campaign) {
+        if (campaign.name) pageview.campaignName = campaign.name;
+        if (campaign.source) pageview.campaignSource = campaign.source;
+        if (campaign.medium) pageview.campaignMedium = campaign.medium;
+        if (campaign.content) pageview.campaignContent = campaign.content;
+        if (campaign.term) pageview.campaignKeyword = campaign.term;
+      }
+
+      var payload = {
+        page: pagePath,
+        title: pageTitle,
+      };
+
+      var resetCustomDimensions = {};
+      for (var i = 0; i < this.resetCustomDimensionsOnPage.length; i++) {
+        var property = this.resetCustomDimensionsOnPage[i];
+        if (dimensionsArray[property]) {
+          resetCustomDimensions[dimensionsArray[property]] = null;
+        }
+      }
+
+      ga("set", resetCustomDimensions);
+      pageview = extend(
```

Comment:
can we update pageview related operations in single flow?
like just after this:
pageview.page = pagePath;
pageview.title = pageTitle;
pageview.location = eventProperties.url;

similarly for payload operations.

Reply:
done